source/user/df.rst

```diff
@@ -5,3 +5,143 @@ Density fitting
+
+Choice of auxiliary basis
+=========================
+
+General consideration
+---------------------
+
+The choice of auxiliary basis depends on the AO basis.
+By default, PySCF uses the pre-defined auxiliary basis set optimized for the AO basis set if the former exists.
+This includes many commonly used AO basis sets in electronic structure calculations, e.g., the Ahlrichs' `def2` family, :cite:`Hellweg07TCA,Weigend98CPL` the Dunning's `cc` family, :cite:`Weigend02PCCP` etc.
+
+When pre-defined auxiliary basis set is not available, an even-tempered basis (ETB) set is generated by the following rule
+
+.. math::
+
+    \varphi &= r^l \exp(-\zeta_{il} r^2), \quad i = 0, 1, \cdots, n \\
+    \zeta_{il} &= \alpha_l \times \beta^i
+
+where :math:`\alpha_l` and :math:`N` are inferred automatically from the AO basis, and :math:`\beta = 2.0` by default.
+
+The user can overwrite the default choice by setting :attr:`auxbasis` upon initialization or at a later stage::
+
+    from pyscf import gto, scf, df
+    mol = gto.M(atom='N 0 0 0; N 0 0 1.2', basis='cc-pvdz')
+    mf = scf.RHF(mol).density_fit(auxbasis="weigend")
+    mf.kernel() # -108.910953335055
+    mf.with_df.auxbasis = "cc-pvdz-jkfit"   # this is the default
+    mf.kernel() # -108.913710743723
+    mf.with_df.auxbasis = df.aug_etb(mol, beta=1.7) # ETB with beta = 1.7
+    mf.kernel() # -108.914059329528
+
+The user can check what auxiliary basis set is used at any time by::
+
+    print(mf.with_df.auxmol.basis)
+
+More examples on using customized auxiliary basis can be found in :source:`examples/df/01-auxbasis.py`.
+
+
+Special consideration for DFT
+-----------------------------
+
+For DFT calculations with pure exchange-correlation functionals (i.e., LDA and GGA), the default auxiliary basis, which is designed for fitting both the Coulomb and the Hartree-Fock exchange integrals, may be unnecessarily too large. We recommend using the ``weigend`` family :cite:`Weigend06PCCP` for a more cost-effective choice as shown in the following example::
+
+    from pyscf import gto, dft
+    mol = gto.M(atom='N 0 0 0; N 0 0 1.2', basis='def2-tzvpp')
+    mf = dft.RKS(mol, xc="pbe").density_fit().run() # -109.432329411505
+    print(mf.with_df.auxmol.basis)                  # default: def2-tzvpp-jkfit
+    print(mf.with_df.auxmol.nao_nr())               # 154 aux basis functions
+    mf.with_df.auxbasis = "weigend"
+    mf.kernel()                                     # -109.432334646585
+    print(mf.with_df.auxmol.nao_nr())               # 98 aux basis functions
+
+
+.. _save_reuse_df_integrals:
+
+Saving and reusing DF integrals
+===============================
+
+By default, the DF integrals are discarded after the calculations are done. Sometimes it is beneficial to save those integrals on disk and re-use them for later calculations. This can be done by specifying a HDF5 file where the DF integrals will be saved via setting :attr:`_cderi_to_save` either at the SCF stage::
```

Comment:
```suggestion
By default, the DF integrals are discarded once the calculation finishes.
Sometimes it is useful to save the integrals on disk for re-use in later calculations.
This can be achieved by specifying a HDF5 file to save the DF integrals in by setting :attr:`_cderi_to_save` either at the SCF stage::
```

Reply:
Are these the DF two- and three-center integrals, or the B-matrices?

Reply:
it's the 3-index cholesky decomposed electron repulsion integrals, CDERIs as it's called in PySCF (which is your B-matrices I believe).
added text explaining this.

Reply:
Yes; B-matrices factorize the integrals `(uv|rs) = sum_P B_{uv}^P B_{rs}^P`, but density fitted B matrices are still not the same as true Cholesky since the former have a fitting error while the latter can be made exact.

It would be cool to have the exact Cholesky decomposition in PySCF...